src/jspdf.js

```diff
@@ -4850,15 +4850,22 @@ function jsPDF(options) {
    * @param {string} postScriptName PDF specification full name for the font.
    * @param {string} id PDF-document-instance-specific label assinged to the font.
    * @param {string} fontStyle Style of the Font.
+   * @param {number || string} fontWeight Weight of the Font.
    * @param {Object} encoding Encoding_name-to-Font_metrics_object mapping.
    * @function
    * @instance
    * @memberof jsPDF#
    * @name addFont
    * @returns {string} fontId
    */
-  API.addFont = function(postScriptName, fontName, fontStyle, encoding) {
-    encoding = encoding || "Identity-H";
+  API.addFont = function(postScriptName, fontName, fontStyle, fontWeight, encoding) {
+    encoding = encoding || "Identity-H"
+    if ((fontStyle == 'normal' && fontWeight === 'bold') || (fontStyle == 'bold' && fontWeight == 'normal')) {
+      throw new Error("Invalid Combination of fontweight and fontstyle");
+    }
+    if (fontWeight && fontStyle !== fontWeight) {
+      fontStyle = fontWeight == 400 ? 'normal' : fontWeight == 700 ? 'bold' : fontStyle + ' ' + fontWeight;
```

Comment:
* make sure to also compare with 'normal' and 'bold' _strings_ as `fontWeight` values.
* Maybe add a comment that the `==` vs `===` is intentional